proxy/http2/Http2Stream.cc

```diff
@@ -356,8 +356,6 @@ Http2Stream::transaction_done()
 
   if (closed) {
     // Safe to initiate SSN_CLOSE if this is the last stream
-    if (cross_thread_event)
-      cross_thread_event->cancel();
     // Schedule the destroy to occur after we unwind here.  IF we call directly, may delete with reference on the stack.
```

Comment:
This seems like a reasonable cleanup unless the earlier do_io_close could cause the cross_thread_event to be set again.  NULLing out cross_thread_event earlier and adding an assert here would be reasonable.

However this fix doesn't seem to match the title.  Canceling the event is not going to cause the close to happen twice.